### DISCUSSION TITLE
[Feat Req] Add support for Ceph as external image store

### DISCUSSION THREAD

**Discussion by wgrcunha** [grafana/grafana#36986] [General]
- Grafana: 4.1.1

I tried to setup external images on Ceph, but i get this error:
t=2017-01-19T19:50:36+0000 lvl=eror msg="Failed to upload alert panel image." logger=alerting.notifier error="unwanted http status 403: \"<?xml version=\\\"1.0\\\" encoding=\\\"UTF-8\\\"?><Error><Code>SignatureDoesNotMatch</Code></Error>\""

Looking at the code, i found this line to get the bucket name:
rBucket := regexp.MustCompile(`https?:\/\/(.*)\.s3(-([^.]+))?\.amazonaws\.com\/?`)

So, its hard coded to use S3 from amazonaws, its possible to change to use any S3 storage service (example ceph)?

Thanks

**Comment by torkelo** [comment]
not sure if the s3 sdk supports it, if it does then maybe.

**Comment by mtanda** [comment]
https://github.com/aws/aws-sdk-go/issues/400
The sdk doesn't support Ceph...
It might be better to implement Ceph native uploader.

**Comment by jpds** [comment]
One can most certainly use the aws-sdk-go library to talk to Ceph, I've used this in the past with Vault's implementation: https://github.com/hashicorp/vault/tree/master/physical/s3

**Comment by ljagiello** [comment]
@jpds my PR (https://github.com/grafana/grafana/pull/7341) use that.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
